src/Util/CapabilitiesUtil/CapabilitiesUtil.js

```diff
@@ -0,0 +1,61 @@
+import OlWMSCapabilities from 'ol/format/wmscapabilities';
+import OlSourceTileWms from 'ol/source/tilewms';
+import OlLayerTile from 'ol/layer/tile';
+
+import { get } from 'lodash';
+
+/**
+ * Helper Class to parse capabilities of WMS layers
+ *
+ * @class CapabilitiesUtil
+ */
+class CapabilitiesUtil {
+
+  /**
+   * @static parseWmsCapabilities - function
+   *
+   * @param {String} capabilitiesUrl Url to WMS capabilities document
+   *
+   * @return {Object} An object representing the WMS capabilities.
+   */
+  static parseWmsCapabilities(capabilitiesUrl) {
+    return fetch(capabilitiesUrl)
+      .then((response) => response.text())
+      .then((data) => {
+        const wmsCapabilitiesParser = new OlWMSCapabilities();
+        return wmsCapabilitiesParser.read(data);
+      });
+  }
+
+  /**
+   * @static getLayersFromCapabilties - parse {OlLayerTile} from capabilities object
+   *
+   * @return {OlLayerTile[]} Array of OlLayerTile
+   */
+  static getLayersFromWmsCapabilties(capabilities) {
+    const wmsVersion = get(capabilities,'version');
+    const wmsAttribution = get(capabilities,'Service.AccessConstraints');
+    const layersInCapabilities = get(capabilities,'Capability.Layer.Layer');
+    const wmsGetMapConfig = get(capabilities, 'Capability.Request.GetMap');
+    const wmsGetFeatureInfoConfig = get(capabilities, 'Capability.Request.GetFeatureInfo');
+    const getMapUrl = get(wmsGetMapConfig,'DCPType[0].HTTP.Get.OnlineResource');
+    const getFeatureInfoUrl = get(wmsGetFeatureInfoConfig,'DCPType[0].HTTP.Get.OnlineResource');
+    return layersInCapabilities.map((layerObj) => new OlLayerTile({
+      opacity: 1,
+      title: get(layerObj, 'Title'),
+      abstract: get(layerObj, 'Attribution'),
+      getFeatureInfoUrl: getFeatureInfoUrl,
+      queryable: !!getFeatureInfoUrl,
+      source: new OlSourceTileWms({
```

Comment:
Always tiled?

Reply:
I've changed this to `OlSourceImageWMS`. IMHO we should return tiled layers in a further method which parses capabilities of WMTS services.